test/ci_workflow_gen.go

```diff
@@ -151,6 +179,81 @@ func parseList(csvList string) []string {
 	return list
 }
 
+func generateSelfHostedUnitTestWorkflows() error {
+	platforms := parseList(unitTestSelfHostedDatabases)
+	for _, platform := range platforms {
+		directoryName := fmt.Sprintf("unit_test_%s", platform)
+		test := &selfHostedTest{
+			Name:              fmt.Sprintf("Unit Test Self Hosted (%s)", platform),
+			ImageName:         fmt.Sprintf("unit_test_%s", platform),
+			Platform:          platform,
+			directoryName:     directoryName,
+			Dockerfile:        fmt.Sprintf("./.github/dockerFiles/%s/Dockerfile", directoryName),
+			MakeTools:         true,
+			InstallXtraBackup: false,
+		}
+		err := setupTestDockerFile(test)
+		if err != nil {
+			return err
+		}
+		filePath := fmt.Sprintf("%s/unit_test_%s_self_hosted.yml", workflowConfigDir, platform)
+		err = writeFileFromTemplate(unitTestSelfHostedTemplate, filePath, test)
+		if err != nil {
+			log.Print(err)
+		}
```

Comment:
Should we return the error here?

Reply:
Actually no. That is the error that we got when we were trying to write self-hosted unit test. In case of an error we still want to continue with the other files